[visionOS] Update SDK additions for visionOS 2.4

emw-apple opened the pull request:
#### c2e3b7aa037524a35192993d83c7a58f23fb060d
<pre>
[visionOS] Update SDK additions for visionOS 2.4
<a href="https://rdar.apple.com/151498066">rdar://151498066</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=293163">https://bugs.webkit.org/show_bug.cgi?id=293163</a>

Reviewed by Jonathan Bedard.

Result of running `extract-tbds-from-internal-sdk --visionos-device`
and `extract-tbds-from-internal-sdk --visionos-simulator` with an
internal debug build directory for device and simulator respectively.

* WebKitLibraries/SDKs/xros2.0-additions.sdk/System/Library/PrivateFrameworks/BackBoardServices.framework/BackBoardServices.tbd:
* WebKitLibraries/SDKs/xros2.0-additions.sdk/System/Library/PrivateFrameworks/CoreRE.framework/CoreRE.tbd:
* WebKitLibraries/SDKs/xros2.0-additions.sdk/System/Library/PrivateFrameworks/LinearMediaKit.framework/LinearMediaKit.tbd:
* WebKitLibraries/SDKs/xros2.0-additions.sdk/System/Library/PrivateFrameworks/MRUIKit.framework/MRUIKit.tbd:
* WebKitLibraries/SDKs/xros2.0-additions.sdk/System/Library/PrivateFrameworks/WritingTools.framework/WritingTools.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator2.0-additions.sdk/System/Library/PrivateFrameworks/BackBoardServices.framework/BackBoardServices.tbd:
* WebKitLibraries/SDKs/xrsimulator2.0-additions.sdk/System/Library/PrivateFrameworks/CoreRE.framework/CoreRE.tbd:
* WebKitLibraries/SDKs/xrsimulator2.0-additions.sdk/System/Library/PrivateFrameworks/LinearMediaKit.framework/LinearMediaKit.tbd:
* WebKitLibraries/SDKs/xrsimulator2.0-additions.sdk/System/Library/PrivateFrameworks/MRUIKit.framework/MRUIKit.tbd:
* WebKitLibraries/SDKs/xrsimulator2.0-additions.sdk/System/Library/PrivateFrameworks/WritingTools.framework/WritingTools.tbd: Added.

Canonical link: <a href="https://commits.webkit.org/295182@main">https://commits.webkit.org/295182@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/523e2a43dceffe7c051fd726882ffdf139bf18e4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/104197 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/23901 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/14225 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/109395 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/54865 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/106237 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/24269 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/32445 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/79140 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/107203 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/18857 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/94007 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/59467 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/18666 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/12044 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/54225 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/88371 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/12103 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/111779 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/31353 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/23117 "Found 2 new test failures: http/tests/navigation/ping-attribute/cross-site-url-anchor-cookie-preexisting-cookie.html http/tests/navigation/ping-attribute/cross-site-url-anchor-cookie.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/88161 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/31717 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/90195 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/87822 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/22383 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/32715 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/10467 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/25817 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/31282 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/31076 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/34412 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/32636 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

megangardner reviewed (commented):
Fixes the VisionPro build.

emw-apple committed:
[visionOS] Update SDK additions for visionOS 2.4
rdar://151498066
https://bugs.webkit.org/show_bug.cgi?id=293163

Reviewed by Jonathan Bedard.

Result of running `extract-tbds-from-internal-sdk --visionos-device`
and `extract-tbds-from-internal-sdk --visionos-simulator` with an
internal debug build directory for device and simulator respectively.

* WebKitLibraries/SDKs/xros2.0-additions.sdk/System/Library/PrivateFrameworks/BackBoardServices.framework/BackBoardServices.tbd:
* WebKitLibraries/SDKs/xros2.0-additions.sdk/System/Library/PrivateFrameworks/CoreRE.framework/CoreRE.tbd:
* WebKitLibraries/SDKs/xros2.0-additions.sdk/System/Library/PrivateFrameworks/LinearMediaKit.framework/LinearMediaKit.tbd:
* WebKitLibraries/SDKs/xros2.0-additions.sdk/System/Library/PrivateFrameworks/MRUIKit.framework/MRUIKit.tbd:
* WebKitLibraries/SDKs/xros2.0-additions.sdk/System/Library/PrivateFrameworks/WritingTools.framework/WritingTools.tbd: Added.
* WebKitLibraries/SDKs/xrsimulator2.0-additions.sdk/System/Library/PrivateFrameworks/BackBoardServices.framework/BackBoardServices.tbd:
* WebKitLibraries/SDKs/xrsimulator2.0-additions.sdk/System/Library/PrivateFrameworks/CoreRE.framework/CoreRE.tbd:
* WebKitLibraries/SDKs/xrsimulator2.0-additions.sdk/System/Library/PrivateFrameworks/LinearMediaKit.framework/LinearMediaKit.tbd:
* WebKitLibraries/SDKs/xrsimulator2.0-additions.sdk/System/Library/PrivateFrameworks/MRUIKit.framework/MRUIKit.tbd:
* WebKitLibraries/SDKs/xrsimulator2.0-additions.sdk/System/Library/PrivateFrameworks/WritingTools.framework/WritingTools.tbd: Added.

Canonical link: https://commits.webkit.org/295182@main
